core/cache/application.go

```diff
@@ -10,7 +10,8 @@ import (
 )
 
 const (
-	applicationConfigChange = "application-config-change"
+	applicationConfigChange   = "application-config-change"
+	applicationCharmURLChange = "application-charmurl-change"
```

Comment:
I would be inclined to name this `modelCharmURLChange` to indicate the nature of the broadcast.

Reply:
i'm on the fence with this one.  it's also generic enough it could be used by another watcher for a single application?  i published with the model to avoid multiple watchers in what's going on here.